spec/acceptance/httpclient/httpclient_spec.rb

```diff
@@ -113,13 +113,8 @@ def filter_response(request, response)
         nil # to let the request be made for real
       end
 
-      # To make two requests that have the same request signature, the headers must match.
-      # Since the webmock server has a Set-Cookie header, the 2nd request will automatically
-      # include a Cookie header (due to how httpclient works), so we have to set the header
-      # manually on the first request but not on the 2nd request.
-      http_request(:get, webmock_server_url, :client => client,
-                         :headers => { "Cookie" => "bar=; foo=" })
-      http_request(:get, webmock_server_url, :client => client)
+      http_request(:get, webmock_server_url, :client => client, :headers => { "Cookie" => "bar=; foo=" })
```

Comment:
So it looks like httpclient made changes around exactly this functionality in 2.6.0
https://github.com/nahi/httpclient/blob/master/CHANGELOG.md

So I am still trying to grok all the changes, reasons, implications etc. And see if we can use 2.6.0 for all ruby versions.